src/connectors/menu/connectMenu.ts

```diff
@@ -115,30 +139,33 @@ export default function connectMenu(renderFn, unmountFn = noop) {
       );
     }
 
-    let sendEvent;
+    let sendEvent: MenuRendererOptions['sendEvent'] | undefined;
+    let _createURL: MenuRendererOptions['createURL'] | undefined;
+    let _refine: MenuRendererOptions['refine'] | undefined;
 
     // Provide the same function to the `renderFn` so that way the user
     // has to only bind it once when `isFirstRendering` for instance
+    let isShowingMore = false;
     let toggleShowMore = () => {};
+    function createToggleShowMore(
+      renderOptions: RenderOptions,
+      widget: Widget
```

Comment:
passing the "this" to this function is the most natural method I found for show more. refinementList and hierarchicalMenu aren't yet converted, so this is the first widget with this pattern